### PR TITLE
Best effort new edit forms / timer popup

### PR DESCRIPTION
**Pull request by tcrammond**
Please remember the [Contributing Guidelines](https://github.com/toggl/toggl-button/blob/master/.github/CONTRIBUTING.md) :heart:

## :star2: What does this PR do?

Updates the appearance/layout of:
* "Edit TE" screen (when clicking on running TE in the toolbar pop up)
* "Timer popup" pop up when you start a timer in an integration

Adds new features:
* When changing the duration in "Edit TE" screen, you may now use `human inputs` in the same way as webapp. **Probably some minor glitches here, please test.**

Known things that won't be fixed:
* Would love to change the tags popdown to be a list (like webapp) and show selected tags at the start, but we're out of time. Will make a feature out of it in future.
* I have noticed that text rendering seems to get butchered slightly in the Chrome browser popup *(compare: open settings, change url to `popup.html`)* - no idea how to fix it.

Screen Shot 2019-04-26 at 11 05 02

Screen Shot 2019-04-26 at 11 06 44

Screen Shot 2019-04-26 at 11 06 52

Screen Shot 2019-04-26 at 11 07 10

## :bug: Recommendations for testing

All changes should be tested across Chrome and Firefox.

Functionality should be.. functional. Check that nothing has gone missing.

Popdowns function sensibly *(they do still miss some keyboard features from webapp etc.)*, duration input functions.

## :memo: Links to relevant issues or information

#1291

### CODE REVIEW

File: src/scripts/background.js [hunk 1]
```diff
@@ -1986,6 +2011,9 @@ window.TogglButton = {
       console.log('Load content script: [' + file + ']');
     }
     browser.tabs.insertCSS(tabId, { file: 'styles/style.css' })
+      .then(function () {
+        return browser.tabs.insertCSS(tabId, { file: 'styles/edit-form.css' });
+      })
       .then(function () {
         browser.tabs.insertCSS(tabId, { file: 'styles/autocomplete.css' });
```
Comment (shantanuraj) [inline src/scripts/background.js]: For consistency and correcting the promise chain I'd recommend retuning here as well
```suggestion
        return browser.tabs.insertCSS(tabId, { file: 'styles/autocomplete.css' });
```

File: package.json [hunk 2]
```diff
@@ -93,6 +93,9 @@
     "@emotion/styled": "^10.0.10",
     "date-fns": "^1.30.1",
     "keycode": "^2.2.0",
+    "lodash.map": "^4.6.0",
```
Comment (shantanuraj) [inline package.json]: Based on the usage we can probably use the native `map/reduce` instead
to save some bits on the package size

**Comment by tcrammond** [comment]
> not sure on the human input part or maybe its a problem in how I'm reviewing it

It's having problems when anything other than `blur` happens. Sorting it 👌

**Comment by TheSoloJake** [comment]
@bryanph Updated your above comment to reflect the correct Jake :)

**Comment by SuperGeekJake** [comment]
We used `react-maskedinput`. See toggl/webapp#9980

**Comment by tcrammond** [comment]
Thx, working on the tweaks. :+1:

About some of the bonus remarks:

> Might be nice to add a back button to opt out of editing the TE

Not adding more scope to this PR but will definitely make an issue for this feature to be added soon.

> Make sure only numbers can be inserted in the time field

It allows flexible "human duration inputs" now, as in webapp, so this isn't necessary. 

> In the time input field, make sure that backspace just resets the number under the cursor to "0"  .....

Would be lovely, but far too out of scope for the goals here *(and it's definitely not a quick pull-in-the-library fix; this form is far from using React 😭)*. Let's put it on the "future UX redesign" list.

**Comment by tcrammond** [comment]
Thanks @bryanph, if you just find a moment to verify the scroll bars issue in Linux has been resolved, that would be great.

File: src/scripts/lib/autocomplete.js [hunk 3]
```diff
@@ -458,6 +459,8 @@ ProjectAutoComplete.prototype.setProjectBullet = function (pid, tid, el) {
 
   let task;
 
+  console.log(pid, tid, el, elem);
```
Comment (shantanuraj) [inline src/scripts/lib/autocomplete.js]: Unguarded log

**Comment by tcrammond** [comment]
> After creating tags from within the tag popdown, saving, editing the TE again and opening the tag popdown again, the created tags are not in the list

Exists here https://github.com/toggl/toggl-button/issues/939 *(2017 😬 )*, thanks for raising it anew. Will schedule it to be worked on.

**Pull request merged**